src/Storage/Device/S3.php

```diff
@@ -530,8 +664,9 @@ private function call(string $method, string $uri, string $data = '')
         \curl_close($curl);
 
         // Parse body into XML
-        if (isset($response->headers['content-type']) && $response->headers['content-type'] == 'application/xml') {
+        if ((isset($response->headers['content-type']) && $response->headers['content-type'] == 'application/xml') || (str_starts_with($response->body, '<?xml'))) {
             $response->body = \simplexml_load_string($response->body);
+            $response->body = json_decode(json_encode($response->body), true);
```

Comment:
I think this is weird enough to warrant a comment - why are we encoding and immediately decoding the response body?

Reply:
The `simple_load_string` converts XML into an object, so i'm encoding the object then decoding into associative array, is there a better way to do this?